go-controller/pkg/clustermanager/network_cluster_controller.go

```diff
@@ -146,7 +146,7 @@ func (ncc *networkClusterController) init() error {
 		return err
 	}
 
-	if ncc.hasNodeAllocation() {
+	if ncc.hasNodeAllocation() || (util.IsNetworkSegmentationSupportEnabled() && ncc.IsPrimaryNetwork()) {
```

Comment:
@tssurya TODO: double check whether we need this in order to get the network-id node allocation for L2 networks.

Reply:
After testing it network-id is fine without this, we can remove this part from the PR.